library/core/src/ptr/const_ptr.rs

```diff
@@ -463,6 +464,21 @@ impl<T: ?Sized> *const T {
     #[stable(feature = "pointer_byte_offsets", since = "1.75.0")]
     #[rustc_const_stable(feature = "const_pointer_byte_offsets", since = "1.75.0")]
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
+    #[requires(
+        // If count is zero, any pointer is valid including null pointer.
+        (count == 0) ||
+        // Else if count is not zero, then ensure that adding `count` doesn't cause 
+        // overflow and that both pointers `self` and the result are in the same 
+        // allocation 
+        (mem::size_of_val_raw(self) != 0 &&
+            (self.addr() as isize).checked_add(count).is_some() &&
+            kani::mem::same_allocation(self, self.wrapping_byte_offset(count)))
+    )]
+    #[ensures(|result|
+        // The resulting pointer should either be unchanged or still point to the same allocation
+        (self.addr() == (*result).addr()) ||
```

Comment:
Shouldn't this also be tied to `count == 0`?

Reply:
@stogaru FYI

Reply:
@xsxszab, please merge my changes into your branch

Reply:
These issues have been addressed in the base branch (PR #169) for this PR. However, since PR 169 is expected to be updated frequently, repeatedly merging its changes into this PR would be inefficient and could introduce duplicate code changes, complicating the review process. Therefore, this PR will be temporarily marked as a draft until PR #169 stabilizes or is merged.

Reply:
Since PR #169 has been merged, I’m undrafting this PR—it’s now ready for review. Note that PR #168 mistakenly removed proofs for unit, integer, composite, and slice types for `add`, `sub` and `offset`. I’ve reverted those changes as part of this PR during merging.